hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -224,22 +198,28 @@ protected CarbonTable getOrCreateCarbonTable(Configuration configuration) throws
 
     // do block filtering and get split
     List<InputSplit> splits =
-        getSplits(job, filterInterface, filteredSegmentToAccess, matchedPartitions, partitionInfo,
-            null, updateStatusManager);
+         getSplits(job, filterInterface, filteredSegmentToAccess, matchedPartitions, partitionInfo,
+            null, updateStatusManager, readCommitted);
+
     // pass the invalid segment to task side in order to remove index entry in task side
-    if (invalidSegments.size() > 0) {
-      for (InputSplit split : splits) {
-        ((org.apache.carbondata.hadoop.CarbonInputSplit) split).setInvalidSegments(invalidSegments);
-        ((org.apache.carbondata.hadoop.CarbonInputSplit) split)
-            .setInvalidTimestampRange(invalidTimestampsList);
+    if (readCommitted instanceof TableStatusReadCommitted) {
```

Comment:
One inputformat will read table according to one readCommitted.  So make it member variable of inputformat

Reply:
done.